libraries/logging/include/logging_stack.h

```diff
@@ -33,30 +33,50 @@
 
 /* FreeRTOS Include. */
 #include "FreeRTOS.h"
+#include "iot_logging_task.h"
 
 /* Standard Include. */
 #include <stdint.h>
 #include <string.h>
 
-/* Metadata information to prepend to every log message. */
-
-/* Macro to extract only the file name from file path to use for metadata in
- * log messages. */
-#ifdef _MSC_VER
-    #define FILENAME           ( strrchr( __FILE__, '\\' ) ? strrchr( __FILE__, '\\' ) + 1 : __FILE__ )
-#else
-    #define FILENAME           ( strrchr( __FILE__, '/' ) ? strrchr( __FILE__, '/' ) + 1 : __FILE__ )
+/**
+ * @brief This config enables extra verbosity in log messages with metadata information
+ * about the source library and location of the log message.
+ * This requires that the toolchain support C99 (for variadic macros) and the GNU
+ * extension for comma elision in variadic macros (with ##__VA_ARGS__).
+ * If this flag is enabled, you can change the metadata information from the logging_stack.h
+ * file.
+ *
+ * @note By default, this configuration is disabled by
+ */
+#ifndef LOGGING_ENABLE_METADATA_WITH_C99_AND_GNU_EXTENSION
+    #define LOGGING_ENABLE_METADATA_WITH_C99_AND_GNU_EXTENSION    0
 #endif
 
-#define LOG_METADATA_FORMAT    "[%s:%d] "
-#define LOG_METADATA_ARGS      FILENAME, __LINE__
+#if LOGGING_ENABLE_METADATA_WITH_C99_AND_GNU_EXTENSION == 1
 
-/* Common macro for all logging interface macros. */
-#if !defined( DISABLE_LOGGING )
-    #define SdkLog( string )    vLoggingPrintf string
-#else
-    #define SdkLog( string )
-#endif
+/* Macro to extract only the file name from file path to use for metadata in
+ * log messages. */
+    #if defined( _MSC_VER_ ) || defined( __ARMCC_VERSION )
+        #define FILENAME    ( strrchr( __FILE__, '\\' ) ? strrchr( __FILE__, '\\' ) + 1 : __FILE__ )
+    #else
+        #define FILENAME    ( strrchr( __FILE__, '/' ) ? strrchr( __FILE__, '/' ) + 1 : __FILE__ )
+    #endif
+
+    #define SdkLogError( message )           SdkLogErrorC99 message
+    #define SdkLogErrorC99( format, ... )    vLoggingPrintf( "[ERROR] [%s] [%s:%d] " format "\r\n", LIBRARY_LOG_NAME, FILENAME, __LINE__, ## __VA_ARGS__ )
```

Comment:
I think it will be better to keep the metadata format and metadata as another MACRO similar to how it was. This will have Single source of truth for metadata information.